<a href="https://colab.research.google.com/github/jaimeload/HW_Terver/blob/main/hw5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
from scipy import stats

##1
Когда используется критерий Стьюдента, а когда Z –критерий?

    Z–критерий используется в случае большого количества наблюдаемых значений
      и когда известна сигма генеральной совокупности, а

    критерий Стьюдента - в противном случае, когда выборка небольшая и дисперсия неизвестна

##2
Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные
автоматическим станком, имеют средний диаметр 17 мм.

Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия генеральной совокупности известна и равна 4 кв. мм.

    H₀: μ = 17
    H₁: μ > 17 ПКО
    α = 0.05
    




In [2]:
X = 17.5
μ = 17
σ = 2
n = 100

In [3]:
tn = (X - μ) / (σ / np.sqrt(n))
tn

2.5

In [4]:
α = 0.05
tk = stats.t.ppf(1 - α, df = n - 1)
tk

1.6603911559963895

    tn > tk ==> H₀ отвергается, а значит что на уровне значимости α = 0.05 есть достаточно
    доказательств, чтобы утверждать, что действительный средний диаметр шариков отличается от 17 мм.

##3
Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья
составляет 200 г.

Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190.

Известно, что их веса распределены нормально.

Верно ли утверждение продавца, если учитывать, что уровень значимости 1%? (Провести двусторонний тест.)

    H₀: μ = 200
    H₁: μ != 200 ДКО
    α = 0,01

In [5]:
μ = 200
n = 10
p = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

X = np.mean(p)
σ = np.std(p, ddof=1)
X, σ

(198.5, 4.453463071962462)

In [7]:
tn = (X - μ) / (σ / np.sqrt(n))
tn

-1.0651074037450896

In [6]:
α = 0.01
tk1 = stats.t.ppf(α / 2, df = n - 1)
tk2 = stats.t.ppf(1 - α / 2, df = n - 1)
tk1, tk2

(-3.24983554401537, 3.2498355440153697)

    tn не попадает в критическую область tk ==> H₀ не отвергается, а значит на уровне значимости α = 0.01
    недостаточно доказательств, чтобы утверждать, что средний вес пачки печенья отличается от 200 г.

##4
Есть ли статистически значимые различия в среднем росте матерей и
дочерей?

Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165

Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160, 163

    H₀: μ_m = μ_d
    H₁: μ_m != μ_d
    α = 0.05

In [8]:
m = np.array([172, 177, 158, 170, 178,175, 164, 160, 169, 165])
d = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160, 163])

In [9]:
stats.ttest_ind(m, d)

TtestResult(statistic=0.4138411497680053, pvalue=0.68387966499013, df=18.0)

    0.68 > 0.05
    p-value > α ==> H₀ не отвергается, а значи нет статистически значимых
    различий в среднем росте матерей и дочерей на уровне значимости α